File: sangsoo/0x0C Backtracking/15650.py
# N과 M(2) Combination
N, M = map(int, input().split())
arr = [0 for _ in range(M)]
isused = [0 for _ in range(N)]

def func(k: int, st: int)->None:
    if (k == M):
        for i in range(M):
            print(arr[i], end = ' ')
        print("")
        return
    for i in range(st, N):
        if (not isused[i]):
            isused[i] = 1
            arr[k] = i+1
            func(k+1, i+1) # 다음번 숫자부터 돌도록. i+1을 넣어줌
            isused[i] = 0
func(0, 0)

File: sangsoo/0x0C Backtracking/15650.cpp
// N과 M(2) Combination
#include <iostream>
int N, M;
int arr[10];
int isused[10];
using namespace std;
void func(int k, int st){
    if(k == M){
        for(int i = 0; i < M; i++)
            cout << arr[i] << ' ';
        cout <<'\n';
        return;
    }
    for (int i = st; i < N; i++){
        if(!isused[i]){
            isused[i] = 1;
            arr[k] = i+1;
            func(k+1, i+1);
            isused[i] = 0;
        }
    }
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> N >> M;
    func(0, 0);
}